### PR TITLE
The solution of the 'car repository' problem

### PR DESCRIPTION
**Pull request by sargisEg**

### CODE REVIEW

File: src/main/com/aca/homework/week8/car_repository/Car.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.aca.homework.week8.car_repository;
+
+public class Car {
+    private String vin;
+    private int year;
+    private int ownersCount = 1;
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/car_repository/Car.java:6]: the car might have no owner

File: src/main/com/aca/homework/week8/car_repository/CarRepository.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.aca.homework.week8.car_repository;
+
+public class CarRepository {
+
+    private Car[] cars = new Car[1000];
+    private int point = 0;
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/car_repository/CarRepository.java:6]: maybe pointer is a better name

File: src/main/com/aca/homework/week8/car_repository/CarRepository.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    public Car save(Car car) {
+        for (int i = 0; i < point; i++) {
+            if (cars[i].equals(car)) {
+                cars[i].addOwnersCount();
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/car_repository/CarRepository.java:11]: ?

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/aca/homework/week8/car_repository/CarRepository.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        cars[pointer] = car;
+        pointer++;
+        return cars[pointer -1];
+    }
+
+    public Car findByVin(String vin) {
+        if(pointer == 0)
+            return null;
+        for (Car car : cars) {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/car_repository/CarRepository.java:23]: here you will get a null pointer exception. also, there is no need to check the whole car array. the performance will be bad

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/aca/homework/week8/car_repository/CarRepository.java (hunk 4 above)
Comment (sargisEg) [reply src/main/com/aca/homework/week8/car_repository/CarRepository.java:23]: Why null pointer exception?

**Pull request merged**